hawkmoth/parser.py

```diff
@@ -239,10 +239,29 @@ def _recursive_parse(comments, cursor, nest, compat):
 
     return [(doc, meta)]
 
+def extract_filename_line(source_loc):
+    valid = re.compile(r".*file\s'(.*)',\sline\s(\d+).*")
+    match = valid.match(source_loc)
+    filename = ''
+    line = -1
+    if match:
+        filename = match.group(1)
+        line = int(match.group(2))
+
+    return filename, line
+
+def parse_gather_diagnostics(severity, errors, diagnostics):
+    for diag in diagnostics:
+        if diag.severity >= severity:
+            filename, line = extract_filename_line(str(diag.location))
+            errors.extend([(diag.severity, filename, line, diag.spelling)])
```

Comment:
Still don't think we should be passing Clang interfaces up the chain. What does 2 mean for us? We know it's a warning in Clang, but when we add other warnings and errors, are we also going to use 2, 3 and 4? Do we skip 0 and 1 entirely? Feels a bit weird...

I was thinking we could hide Clang's severity levels behind a `class ErrorLevel(Enum)` or something (look into `docstr.py` for an example) we could then leverage everywhere consistently.